reframe/core/launchers/mpi.py

```diff
@@ -48,8 +48,11 @@ def command(self, job):
         if self.use_cpus_per_task and job.num_cpus_per_task:
             ret.append(f'--cpus-per-task={job.num_cpus_per_task}')
 
-        return ret
+        if self.environment_variables:
+            env_vars = ','.join(f'{k}={v}' for k, v in self.environment_variables.items())
```

Comment:
This line is too long but it may fit once you rename `environment_variables` to `env_vars`.